src/nix.rs

```diff
@@ -0,0 +1,338 @@
+                    arg.to_string().into(),
+                    translate(n.body().unwrap(), file_id),
+                ),
+                NODE_PATTERN => {
+                    use crate::destruct::*;
+                    let pat = rnix::types::Pattern::cast(arg).unwrap();
+                    let at = pat.at().map(|id| id.as_str().into());
+                    // TODO: manage default values:
+                    let matches = pat
+                        .entries()
+                        .map(|e| {
+                            let mv = if let Some(def) = e.default() {
+                                MetaValue {
+                                    value: Some(translate(def, file_id)),
+                                    priority: MergePriority::Default,
+                                    ..Default::default()
+                                }
+                            } else {
+                                Default::default()
+                            };
+                            Match::Simple(e.name().unwrap().as_str().into(), mv)
+                        })
+                        .collect();
+                    let dest = Destruct::Record {
+                        matches,
+                        open: pat.ellipsis(),
+                        rest: None,
+                        span,
+                    };
+                    Term::FunPattern(at, dest, translate(n.body().unwrap(), file_id)).into()
+                }
+                _ => unreachable!(),
+            }
+            .into()
+        }
+
+        ParsedType::Apply(n) => Term::App(
+            translate(n.lambda().unwrap(), file_id),
+            translate(n.value().unwrap(), file_id),
+        )
+        .into(),
+        ParsedType::IfElse(n) => Term::App(
+            Term::App(
+                Term::Op1(UnaryOp::Ite(), translate(n.condition().unwrap(), file_id)).into(),
+                translate(n.body().unwrap(), file_id),
+            )
+            .into(),
+            translate(n.else_body().unwrap(), file_id),
+        )
+        .into(),
+        ParsedType::BinOp(n) => (n, file_id).into(),
+        ParsedType::UnaryOp(n) => (n, file_id).into(),
+        ParsedType::Select(n) => match ParsedType::try_from(n.index().unwrap()).unwrap() {
+            ParsedType::Ident(id) => Term::Op1(
+                UnaryOp::StaticAccess(id.as_str().into()),
+                translate(n.set().unwrap(), file_id),
+            )
+            .into(),
+            _ => Term::Op2(
+                BinaryOp::DynAccess(),
+                translate(n.index().unwrap(), file_id),
+                translate(n.set().unwrap(), file_id),
+            )
+            .into(),
+        },
+        ParsedType::OrDefault(n) => unimplemented!(),
+
+        // TODO: what are these?
+        ParsedType::PatBind(_) | ParsedType::PathWithInterpol(_) => {
+            unimplemented!()
+        }
+
+        // Is not a `RichTerm` so is transformed as part of actual ones.
```

Comment:
I didn't understand what this comment mean.

Reply:
Mmmmh... me neither ;)
let me think...
Actualy, these variants are not "Terms" in the sens they can not be parsed independently of a Term. They can only be used in a term. Our `translate` function return a term so these can not be parsed at this level. they are parsed as parts of actual terms.
...
Is it more clear?